README.md

```diff
@@ -229,175 +61,46 @@ The activation can always be verified by running `autohooks check`.
 
+For managing these dependencies currently three modes are supported by
+autohooks:
 
-def get_include(config)
-    if not config:
-        return DEFAULT_INCLUDE
+* `pythonpath` for dependency management via [pip]
+* `poetry` for dependency management via [poetry] (recommended)
+* `pipenv` for dependency management via [pipenv]
 
-    config = config.get('tool', 'autohooks', 'plugins', 'foo')
-    return config.get_value('include', DEFAULT_INCLUDE)
-
-
-def precommit(**kwargs):
-    config = kwargs.get('config')
-    include = get_include(config)
-
-    files = [f for f in get_staged_status() if match(f.path, include)]
-
-    if not files:
-      # not files to lint
-      return 0
-
-    with stash_unstaged_changes(files):
-        const failed = False
-        for file in files:
-            status = subprocess.call(['foolinter', str(file)])
-            if status:
-                fail('Could not validate {str(file)}')
-                failed = True
-            else:
-                ok('Validated {str(file)}')
-
-        return 1 if failed else 0
-```
-
-### Formatting Plugin
-
-Usually the standard call sequence for a formatting plugin is the following:
-
-1. get list of staged files
-2. filter list of files for a specific file type
-3. stash unrelated changes
-4. apply formatting on filtered list of files by calling some external tool
-5. raise exception if something did go wrong
-6. stage changes made by the tool
-7. unstash unrelated changes
-8. return 0
-
-Example plugin:
-
-```python3
-import subprocess
-
-from autohooks.api import ok, error
-from autohooks.api.git import (
-    get_staged_status,
-    stage_files_from_status_list,
-    stash_unstaged_changes,
-)
-from autohooks.api.path import match
-
-DEFAULT_INCLUDE = ('*.ext')
-
-
-def get_include(config)
-    if not config:
-        return DEFAULT_INCLUDE
-
-    config = config.get('tool', 'autohooks', 'plugins', 'bar')
-    return config.get_value('include', DEFAULT_INCLUDE)
-
-
-def precommit(**kwargs):
-    config = kwargs.get('config')
-    include = get_include(config)
-
-    files = [f for f in get_staged_status() if match(f.path, include)]
-
-    if not files:
-      # not files to format
-      return 0
+These modes handle how autohooks, the plugins and their dependencies are loaded
+during git hook execution.
 
-    with stash_unstaged_changes(files):
-        for file in files:
-            # run formatter and raise exception if it fails
-            subprocess.run(['barformatter', str(file)], check=True)
-            ok('Formatted {str(file)}')
+If no mode is specified in the [`pyproject.toml` config file](#configure-mode-and-plugins-to-be-run)
+and no mode is set during [activation](#activating-the-git-hooks), autohooks
+will use the [pythonpath mode](#pythonpath-mode) by default.
 
-        return 0
-```
+For more details on using [pip], [poetry] or [pipenv] in conjunction with these
+modes see the documentation.
```

Comment:
Maybe link it here?

Reply:
Yes of course but I don't have a URL yet 😝